test/jit_hooks/model.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import os
+import sys
+import torch
+
+# grab modules from test_jit_hooks.cpp
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
+from jit.test_hooks_modules import *
+
+
+# Tests for JIT forward hooks and pre-hooks
+def test_module_no_forward_input():
+    m_scripted = torch.jit.script(test_module_no_forward_input_model())
+    m_scripted.save(save_name + "test_module_no_forward_input" + ".pt")
+
+
+def test_module_forward_multiple_inputs():
+    m_scripted = torch.jit.script(test_module_forward_multiple_inputs_model())
+    m_scripted.save(save_name + "test_module_forward_multiple_inputs" + ".pt")
+
+
+def test_module_multiple_hooks_multiple_inputs():
+    m_scripted = torch.jit.script(test_module_multiple_hooks_multiple_inputs_model())
+    m_scripted.save(save_name + "test_module_multiple_hooks_multiple_inputs" + ".pt")
+
+
+def test_module_forward_single_input():
+    m_scripted = torch.jit.script(test_module_forward_single_input_model())
+    m_scripted.save(save_name + "test_module_forward_single_input" + ".pt")
+
+
+def test_module_same_hook_repeated():
+    m_scripted = torch.jit.script(test_module_same_hook_repeated_model())
+    m_scripted.save(save_name + "test_module_same_hook_repeated" + ".pt")
+
+
+def test_module_hook_return_nothing():
+    m_scripted = torch.jit.script(test_module_hook_return_nothing_model())
+    m_scripted.save(save_name + "test_module_hook_return_nothing" + ".pt")
+
+
+def test_module_multiple_hooks_single_input():
+    m_scripted = torch.jit.script(test_module_multiple_hooks_single_input_model())
+    m_scripted.save(save_name + "test_module_multiple_hooks_single_input" + ".pt")
+
+
+def test_submodule_forward_multiple_inputs():
+    m_scripted = torch.jit.script(test_submodule_forward_multiple_inputs_model())
+    m_scripted.save(save_name + "test_submodule_forward_multiple_inputs" + ".pt")
+
+
+def test_submodule_multiple_hooks_multiple_inputs():
+    m_scripted = torch.jit.script(test_submodule_multiple_hooks_multiple_inputs_model())
+    m_scripted.save(save_name + "test_submodule_multiple_hooks_multiple_inputs" + ".pt")
+
+
+def test_submodule_forward_single_input():
+    m_scripted = torch.jit.script(test_submodule_forward_single_input_model())
+    m_scripted.save(save_name + "test_submodule_forward_single_input" + ".pt")
+
+
+def test_submodule_same_hook_repeated():
+    m_scripted = torch.jit.script(test_submodule_same_hook_repeated_model())
+    m_scripted.save(save_name + "test_submodule_same_hook_repeated" + ".pt")
+
+
+def test_submodule_hook_return_nothing():
+    m_scripted = torch.jit.script(test_submodule_hook_return_nothing_model())
+    m_scripted.save(save_name + "test_submodule_hook_return_nothing" + ".pt")
```

Comment:
How come `save` is used here instead of `torch.jit.save`? Importing issues?

Reply:
no reason besides me not knowing the difference between the two